src/components/shared/chart/Canvas.js

```diff
@@ -135,8 +143,18 @@ export default class ChartCanvas<HoveredItem> extends React.Component<
     }
   }
 
-  _onDoubleClick() {
-    this.props.onDoubleClickItem(this.state.hoveredItem);
+  _onDoubleClick(event: SyntheticMouseEvent<>) {
+    if (!this._canvas) {
+      return;
+    }
+
+    // The mouseDownListener in the Viewport triggers state updates
+    // which in turn causes componentWillReceiveProps in this
+    // component to clear the hovered item.  Thus, at the time when
+    // this event handler is called, `this.state.hoveredItem` is null.
+    // Obtain the hovered item from the mouse event instead.
+    const maybeHoveredItem = this._hoveredItemFromMouseEvent(event);
```

Comment:
nit: maybe something like:
```javascript
const maybeHoveredItem = this.state.hoveredItem === null
  ? this._hoveredItemFromMouseEvent(event)
  : this.state.hoveredItem;
```

Note I don't use `this.state.hoveredItem || this._hoveredItemFromMouseEvent(event)` because we don't know what type  `hoveredItem` really is, it could be a number and thus could be `0`.